pgmpy/factors/continuous/JointGaussianDistribution.py

```diff
@@ -0,0 +1,330 @@
+        ...             np.array([[4, 2, -2], [2, 5, -5], [-2, -5, 8]]))
+        >>> dis.precision_matrix
+        array([[ 0.3125    , -0.125     ,  0.        ],
+                [-0.125     ,  0.58333333,  0.33333333],
+                [ 0.        ,  0.33333333,  0.33333333]])
+        """
+
+        if self._precision_matrix is None:
+            self._precision_matrix = np.linalg.inv(self.covariance)
+        return self._precision_matrix
+
+    def marginalize(self, variables, inplace=True):
+        """
+        Modifies the distribution with marginalized values.
+
+        Parameters
+        ----------
+
+        variables: iterator
+                List of variables over which to marginalize.
+        inplace: boolean
+                If inplace=True it will modify the distribution itself,
+                else would return a new distribution.
+
+        Returns
+        -------
+        JointGaussianDistribution or None :
+                if inplace=True (default) returns None
+                if inplace=False return a new JointGaussianDistribution instance
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from pgmpy.factors import JointGaussianDistribution as JGD
+        >>> dis = JGD(['x1', 'x2', 'x3'], np.array([[1], [-3], [4]]),
+        ...             np.array([[4, 2, -2], [2, 5, -5], [-2, -5, 8]]))
+        >>> dis.variables
+        ['x1', 'x2', 'x3']
+        >>> dis.mean
+        array([[ 1],
+                [-3],
+                [ 4]])
+        >>> dis.covariance
+        array([[ 4,  2, -2],
+                [ 2,  5, -5],
+                [-2, -5,  8]])
+
+        >>> dis.marginalize(['x3'])
+        dis.variables
+        ['x1', 'x2']
+        >>> dis.mean
+        array([[ 1],
+                [-3]]))
+        >>> dis.covariance
+        narray([[4, 2],
+               [2, 5]])
+        """
+
+        if isinstance(variables, six.string_types):
+            raise TypeError("variables: Expected type list or array-like, got type str")
+
+        distribution = self if inplace else self.copy()
+
+        var_indexes = [distribution.variables.index(var) for var in variables]
+        index_to_keep = [self.variables.index(var) for var in self.variables
+                         if var not in variables]
+
+        distribution.variables = [distribution.variables[index] for index in index_to_keep]
+        distribution.mean = distribution.mean[index_to_keep]
+        distribution.covariance = distribution.covariance[np.ix_(index_to_keep, index_to_keep)]
+        distribution._precision_matrix = None
+
+        if not inplace:
+            return distribution
+
+    def reduce(self, values, inplace=True):
+        """
+        Reduces the distribution to the context of the given variable values.
+
+        The formula for the obtained conditional distribution is given by -
+
+        For, N(X_j | X_i = x_i) ~ MN(mu_j.i, sig_j.i)
+
```

Comment:
@yashu-seth I think this link would be helpful in writing documentation with maths http://www.sphinx-doc.org/en/stable/ext/math.html